packages/api/src/command/medical/admin-upload-doc.ts

```diff
@@ -0,0 +1,98 @@
+import { v4 as uuidv4 } from "uuid";
+import dayjs from "dayjs";
+import { makeFhirApi } from "../../external/fhir/api/api-factory";
+import { getOrganizationOrFail } from "./organization/get-organization";
+import { getPatientOrFail } from "./patient/get-patient";
+import { Config } from "../../shared/config";
+
+const docContributionUrl = Config.getDocContributionUrl();
+
+export async function createAndUploadDocReference({
+  cxId,
+  patientId,
+  file,
+  metadata,
+}: {
+  cxId: string;
+  patientId: string;
+  file: Express.Multer.File;
+  metadata: {
+    description: string;
+  };
+}): Promise<void> {
+  const patient = await getPatientOrFail({ id: patientId, cxId });
+  const organization = await getOrganizationOrFail({ cxId });
+
+  const fhirApi = makeFhirApi(cxId);
+  const docRefId = uuidv4();
+
+  const now = dayjs();
+
+  const data = `{
+        "resourceType": "DocumentReference",
+        "id": "${docRefId}",
+        "contained": [
+            {
+                "resourceType": "Organization",
+                "id": "${organization.id}",
+                "name": "${organization.data.name}"
+            },
+            {
+                "resourceType": "Patient",
+                "id": "${patient.id}"
+            }
+        ],
+        "masterIdentifier": {
+            "system": "urn:ietf:rfc:3986",
+            "value": "${docRefId}"
+        },
+        "identifier": [
+            {
+                "use": "official",
+                "system": "urn:ietf:rfc:3986",
+                "value": "${docRefId}"
+            }
+        ],
+        "status": "current",
+        "type": {
+          "coding": [
+              {
+                  "system": "http://loinc.org/",
+                  "code": "75622-1",
+                  "display":  "${metadata.description}"
+              }
+          ]
+        },
+        "subject": {
+            "reference": "Patient/${patient.id}",
+            "type": "Patient"
+        },
+        "author": [
+            {
+                "reference": "#${organization.id}",
+                "type": "Organization"
+            }
+        ],
+        "description": "${metadata.description}",
+        "content": [
+            {
+                "attachment": {
+                    "contentType": "${file.mimetype}",
+                    "url": "${docContributionUrl}?fileName=${file.originalname}"
+                }
+            }
+        ],
```

Comment:
I suggest we use the same structure for this as this is a Metriport content - similar to when we add CW-originated doc refs ([link](https://github.com/metriport/metriport/blob/4502bdcd3ad4398061b8f13e6710da98a4fc7990/packages/api/src/external/fhir/document/index.ts#L49C15-L49C15)).

Even better would be to create a function to return a "Metriport DocumentReferenceContent" and reuse here and there.

I'd also check other fields we set there, so we try to be similar, use the same logic (obviously applied to a Metriport-originated doc ref).

Reply:
Both "system" are like namespaces, scopes that determine known values/codes.

From [Identifier](https://www.hl7.org/fhir/R4/datatypes.html#identifier) (masterIdentifier):

> A numeric or alphanumeric string that is associated with a single object or entity within a given system.
> ...
> The `value` SHALL be unique within the defined system and have a consistent meaning wherever it appears.
> ...
> The `system` is a URI that defines a set of identifiers (i.e. how the value is made unique). It might be a specific application or a recognized standard/specification for a set of identifiers or a way of making identifiers unique.

From [Coding](https://www.hl7.org/fhir/R4/datatypes.html#Coding):

> A Coding is a representation of a defined concept using a symbol from a defined "code system"
> ...
> The meaning of the Coding is defined by the `code`. The `system` provides the source of the definition of the code, ...

I guess this is the constraint for IDs (from identifier):

> If the identifier value itself is naturally a globally unique URI (e.g. an OID, a UUID, or a URI with no trailing local part), then the system SHALL be "urn:ietf:rfc:3986"